TensorLib/Basic.lean

```diff
@@ -2,4 +2,98 @@ namespace TensorLib
 
 def hello := "world"
 
+class TensorElement (a: Type) where
+  add: a -> a -> a
```

Comment:
You should use those: https://leanprover-community.github.io/mathlib4_docs/Init/Prelude.html#Add.add
That way, you'll get the nice syntax and the theorems.

Reply:
Ah ok, so we will have type classes for each primitive operation and then mix them together as needed for each more complex operation. Seems nice.